src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/CVObjectDetectionEvaluator.cs

```diff
@@ -0,0 +1,449 @@
+                        RGDebug.LogDebug($"CVObjectDetectionEvaluator - Cleanup - botSegment: {segmentNumber} - abortingWebRequests");
+                        //try to abort the request
+                        foreach (var action in requests.Values)
+                        {
+                            action.Invoke();
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        // DO NOTHING .. we tried.. we really did
+                    }
+                }
+
+                // remove the tracked result
+                _queryResultTracker.Remove(segmentNumber, out _);
+                _priorResultsTracker.Remove(segmentNumber, out _);
+                RGDebug.LogVerbose($"CVObjectDetectionEvaluator - Cleanup - botSegment: {segmentNumber} - END");
+            }
+        }
+
+        // Returns a list of non-matched entries
+        public static List<string> Matched(int segmentNumber, List<KeyFrameCriteria> criteriaList)
+        {
+            RGDebug.LogVerbose($"CVObjectDetectionEvaluator - Matched - botSegment: {segmentNumber} - BEGIN");
+
+            ConcurrentDictionary<int, List<CVObjectDetectionResult>> objectDetectionResults = null;
+            List<string> priorResults = null;
+            bool requestInProgress = false;
+
+            // Try to get the results or request in progress.
+            lock (_requestTracker)
+            {
+                // Check if we have results or a request in progress.
+                requestInProgress = _requestTracker.ContainsKey(segmentNumber);
+
+                // Get the results for the segment.
+                _queryResultTracker.TryGetValue(segmentNumber, out objectDetectionResults);
+                // Get the prior results for the segment.
+                _priorResultsTracker.TryGetValue(segmentNumber, out priorResults);
+
+                // Print the results if there are any.
+                if (objectDetectionResults != null)
+                {
+                    var resultsString = $"\n[{string.Join(",\n", objectDetectionResults.Select(pair => $"{pair.Key}:[{(pair.Value == null ? "null" : string.Join(",\n", pair.Value))}]"))}]";
+                    RGDebug.LogDebug($"CVObjectDetectionEvaluator - Matched - botSegment: {segmentNumber} - cvImageResults: {resultsString}");
+                }
+            }
+
+            var resultList = new List<string>();
+
+            // If no results and no request, we'll initiate a new request.
+            if (!requestInProgress && objectDetectionResults == null)
+            {
+                resultList = TryPostRequest(segmentNumber, criteriaList, resultList);
+            }
+            // If we have results and no request in progress, we'll evaluate them.
+            // We check priorResults null here so that we only evaluate a new result 1 time.
+            else if (!requestInProgress && objectDetectionResults != null && priorResults == null)
```

Comment:
I noticed that this check was slightly different from the check in the other file. Just curious to learn more about that!

Reply:
I see, this check is in the other method now. Thanks!